eli5/formatters/image.py

```diff
@@ -0,0 +1,324 @@
+
+
+def format_as_image(expl, # type: Explanation
+    interpolation=Image.LANCZOS, # type: int
+    colormap=matplotlib.cm.magma, # type: Callable[[np.ndarray], np.ndarray]
+    alpha_limit=0.65, # type: Optional[Union[float, int]]
+    ):
+    # type: (...) -> Image
+    """format_as_image(expl, interpolation=Image.LANCZOS, colormap=matplotlib.cm.magma, alpha_limit=0.65)
+
+    Format a :class:`eli5.base.Explanation` object as an image.
+
+    Note that this formatter requires ``matplotlib`` and ``Pillow`` optional dependencies.
+    
+    Parameters
+    ----------
+    interpolation: int, optional
+        Interpolation ID or Pillow filter to use when resizing the image.
+        
+        Example filters from PIL.Image
+            * ``NEAREST``
+            * ``BOX``
+            * ``BILINEAR``
+            * ``HAMMING``
+            * ``BICUBIC``
+            * ``LANCZOS``
+
+        See also https://pillow.readthedocs.io/en/stable/handbook/concepts.html#filters.
+
+        *Note that these attributes are integer values*.
+
+        Default is ``PIL.Image.LANCZOS``. 
+
+    colormap: callable, optional
+        Colormap scheme to be applied when converting the heatmap from grayscale to RGB.
+        Either a colormap from matplotlib.cm, 
+        or a callable that takes a rank 2 array and 
+        returns the colored heatmap as a [0, 1] RGBA numpy array.
+
+        Example colormaps from matplotlib.cm
+            * ``viridis``
+            * ``jet``
+            * ``binary``
+        
+        See also https://matplotlib.org/gallery/color/colormap_reference.html.
+
+        Default is ``matplotlib.cm.magma`` (blue to red).
+
+    alpha_limit: float or int, optional
+        Maximum alpha (transparency / opacity) value allowed 
+        for the alpha channel pixels in the RGBA heatmap image.
+
+        Between 0.0 and 1.0.
+
+        Useful when laying the heatmap over the original image, 
+        so that the image can be seen over the heatmap.
+
+        Default is 0.65.
+
+    Returns
+    -------
+    overlay : PIL.Image.Image
+        PIL image instance of the heatmap blended over the image.
+    """
+    image = expl.image
+    heatmap = expl.heatmap
+    
+    # We first 1. colorize 2. resize
+    # as opposed 1. resize 2. colorize
+
+    heatmap = colorize(heatmap, colormap=colormap)
+    # TODO: automatically detect which colormap would be the best based on colors in the image
+
+    # make the alpha intensity correspond to the grayscale heatmap values
+    # cap the intensity so that it's not too opaque when near maximum value
+    # TODO: more options for controlling alpha, i.e. a callable?
+    heat_values = expl.heatmap
+    update_alpha(heatmap, starting_array=heat_values, alpha_limit=alpha_limit)
+
+    heatmap = heatmap_to_rgba(heatmap)
+
+    heatmap = resize_over(heatmap, image, interpolation=interpolation)
+    overlay = overlay_heatmap(heatmap, image)
+    return overlay
+
+
+def heatmap_to_grayscale(heatmap):
```

Comment:
Will this function and other public function below appear in the docs? If yes, I think it would be better to  mark all functions which are not in the public API as private (starting with an underscore, or maybe hide them in another way?). For this and similar functions, I think we don't expect the user to call them? Making them private makes API and docs smaller, and allows us to refactor code in the future more freely.

Reply:
Using an underscore works well I think, since the docs don't show such functions automatically.

Reply:
I think `heatmap_to_grayscale()` and `heatmap_to_rgba()` are okay as public functions for convenience? The rest can be hidden except for `format_as_image()`.

Reply:
Yeah, it's fine to keep such helpers public, and I like `expand_heatmap` name